realm/realm-library/src/main/java/io/realm/RealmObjectSchema.java

```diff
@@ -588,11 +588,36 @@ public boolean isEmbedded() {
      * @see RealmClass#embedded()
      */
     public void setEmbedded(boolean embedded) {
+        setEmbedded(embedded, false);
+    }
+
+    /**
+     * Converts the class to be embedded or not.
+     * <p>
+     * A class can only be marked as embedded if the following invariants are satisfied:
+     * <ul>
+     *     <li>
+     *         The class is not allowed to have a primary key defined.
+     *     </li>
+     *     <li>
+     *         All existing objects of this type, must have one and exactly one parent object
+     *         already pointing to it. If 0 or more than 1 object has a reference to an object
+     *         about to be marked embedded an {@link IllegalStateException} will be thrown.
+     *     </li>
+     * </ul>
+     *
+     * This variant adds on {@link RealmObjectSchema#setEmbedded(boolean)} to add support for automatic handling of unsatisfied
+     * invariants after converting the property into an embedded object property.
```

Comment:
We could mention that as part of the automatic handling process, objects might be cloned if required.